crates/bevy_pbr/src/render/double_buffer.rs

```diff
@@ -0,0 +1,79 @@
+use bevy_render::{
+    render_resource::{Buffer, BufferUsages, BufferVec},
+    renderer::{RenderDevice, RenderQueue},
+};
+use bytemuck::Pod;
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum Side {
+    A,
+    B,
+}
+impl Side {
+    fn flip(self) -> Self {
+        match self {
+            Side::A => Side::B,
+            Side::B => Side::A,
+        }
+    }
+    fn toggle(&mut self) {
+        *self = self.flip();
+    }
+}
```

Comment:
I'd have just used a `[BufferVec<T>; 2]` and `index: usize`, and then update the index like `index = (index + 1) & 1;` No need for all this 'Side' stuff. :)

Reply:
I did not consider this. I changed it to use a `[BufferVec<T>; 2]`. It's a bit cleaner.